lib/diagnostic.js

```diff
@@ -6,26 +6,46 @@
 // splits it into an array of normalized words
 // (uppercase strings without punctuation)
 // and returns that array.
-const normalizeWords = function () {
-
+const normalizeWords = function (str) {
+  let normalWords = [];
+  normalWords = str.split(/\s+/);
+  return normalWords;
```

Comment:
A few problems here. First all you have to do with normalWords is ```let normalWords = str.split(/\s+/);``` because the return value of split is an array so no need to declare an empty array beforehand. next you left out the for loop that removes all non word characters and sets all word characters to uppercase. That would look like 

```
for (let i = 0; i < normalWords.length; i++) {
    let word = normalWords[i];
    normalWords[i] = word.replace(/\W+/, '').toUpperCase();
  }
```